documentation/api/query/v4/reports.markdown

```diff
@@ -105,12 +107,41 @@ Resource event objects are of the following form:
       "containment_path": <containment heirarchy of resource within catalog>
     }
 
+The metrics field may be either null, in the case of reports submitted without metrics,
+or a JSON array of objects like so:
+
+
+    {
+      "category" : <category of metric ("resources", "time", "changes", or "events")>,
+      "name" : <name of the metric>,
+      "value" : <value of the metric (double precision)>
+    }
+
+The logs field may be either null, in the case of reports submitted without
+logs, or a JSON array of objects like so, each corresponding to a log line in a
+Puppet run:
+
+   {
+       "file" : <file of resource declaration>,
+       "line" : <line of resource declaration>,
+       "level" : <log level>,
+       "message" : <log message>
+       "source" : <log source>
+       "tags" : [<resource tag>],
+       "time" : <log line timestamp>
+    }
+
+File and line may each be null if the log does not concern a resource.
+
 **Note on fields that allow `NULL` values**
 
 In the resource_event schema above, `containment_path`, `new_value`, `old_value`, `property`, `file`, `line`, `status`, and `message` may all be null.
 
-**Note on querying resource events**
-The `reports` endpoint does not support querying on the value of `resource_events`, but the same information can be be accessed by querying the `events` endpoint for events with field `report` equal to a given report's `hash`.
+**Note on querying resource events, metrics, and logs**
```

Comment:
Extra \n would make this look like the last note. Its a pre-existing problem inconsistency I know.